env.sh

```diff
@@ -1,15 +1,146 @@
 #!/bin/bash
 
-# Shared environment variables.
+function which_flavour {
+    if [[ -f /etc/os-release ]]; then
+        detected="$(grep '^ID=' /etc/os-release | cut -d= -f2)"
+    fi
+    echo "$detected"
+}
+
+case "$OSTYPE" in
+    darwin*)
+        OS='macos'
+        ;;
+    linux*)
+        OS=$(which_flavour)
+        ;;
+    *)
+        OS="not detected"
+        ;;
+esac
+
+echo "OS detected is $OS"
+read -rp "Is it correct ? (y|N)" input
+input=$(echo "$input" | tr "[:lower:]" "[:upper:]")
+case $input in
+    y|yes)
```

Comment:
Because of the upper above
```suggestion
    Y|YES)
```